pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -753,137 +735,167 @@ func (r *ReconcileChaosEngine) removeChaosRunner(engine *chaosTypes.EngineInfo,
 	}
 
 	// Fetch the app details from ChaosEngine instance. Check if app is present
 	// Also check, if the app is annotated for chaos & that the labels are unique
-	engine, err = getApplicationDetail(engine)
+	err = getApplicationDetail(engine)
 	if err != nil {
 		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to update chaosengine")
-		return reconcile.Result{}, err
+		return err
 	}
 
 	if engine.Instance.Spec.AnnotationCheck == "true" {
 		// Determine whether apps with matching labels have chaos annotation set to true
 		engine, err = resource.CheckChaosAnnotation(engine)
 		if err != nil {
-			r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to find chaosengine")
+			r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to get chaosengine")
 			chaosTypes.Log.Info("Annotation check failed with", "error:", err)
-			return reconcile.Result{}, nil
+			return nil
 		}
 	}
-	// Define an engineRunner pod which is secondary-resource #1
-	engineRunner, err := newRunnerPodForCR(*engine)
-	if err != nil {
-		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to create chaos resources")
+	return nil
+}
 
-		return reconcile.Result{}, err
+func (r *ReconcileChaosEngine) updateEngineForComplete(engine *chaosTypes.EngineInfo, isCompleted bool) error {
+	if engine.Instance.Status.EngineStatus != litmuschaosv1alpha1.EngineStatusCompleted {
+		engine.Instance.Status.EngineStatus = litmuschaosv1alpha1.EngineStatusCompleted
+		engine.Instance.Spec.EngineState = litmuschaosv1alpha1.EngineStateStop
+		if err := r.client.Update(context.TODO(), engine.Instance, &client.UpdateOptions{}); err != nil {
+			return err
+		}
+		r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "ChaosEngineCompleted", "Chaos Engine completed, will delete or retain the resources according to jobCleanUpPolicy")
 	}
+	return nil
+}
 
-	//Check if the engineRunner pod already exists, else create
-	engineReconcile, err := r.checkEngineRunnerPod(reqLogger, engineRunner)
-	if err != nil {
-		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to get chaos resources")
-		return reconcile.Result{}, err
-	}
-	// If monitoring is set to true,
-	// Define an engineMonitor pod which is secondary-resource #2 and
-	// Define an engineMonitor service which is secondary-resource #3
-	// in the same namespace as CR
-	reconcileResult, err := checkMonitoring(engineReconcile, reqLogger)
-	if err != nil {
-		r.recorder.Eventf(engine.Instance, corev1.EventTypeWarning, "ChaosResourcesOperationFailed", "Unable to get chaos resources")
-		return reconcileResult, err
+func (r *ReconcileChaosEngine) updateEngineForRestart(engine *chaosTypes.EngineInfo) error {
+	r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "RestartInProgress", "Dancer")
```

Comment:
check the error msg? Dancer?

Reply:
Have asked @ksatchit to provide string for this Event @ispeakc0de .

Reply:
Chaos Experiment <exp-name>re-triggered?  (this assumes we have a 1-1 map) 
Chaos runner <name> recreated (tasks at engine resource level)

Reply:
@ksatchit Its the case when we restart the Engine, delete and recreate all resources.